tests/rkt_volume_mount_test.go

```diff
@@ -0,0 +1,343 @@
+	}
+
+	// mount tmpfs for /dir1/dir2
+	if err := syscall.Mount("", tmpdir2path, "tmpfs", 0, ""); err != nil {
+		executeFuncsReverse(cleanupFuncs)
+		t.Fatalf("Can't mount tmpfs on inner temp directory %q: %v", tmpdir2path, err)
+	}
+	cleanupFuncs = append(cleanupFuncs, func() {
+		if err := syscall.Unmount(tmpdir2path, syscall.MNT_DETACH); err != nil {
+			t.Errorf("could not unmount %q: %v", tmpdir2path, err)
+		}
+	})
+	cleanupFuncs = append(cleanupFuncs, func() { os.RemoveAll(tmpdir) })
+
+	// create the file in dir2 after the mount
+	tmpdir2innerfile, err := os.Create(tmpdir2filepath)
+	if err != nil {
+		executeFuncsReverse(cleanupFuncs)
+		t.Fatalf("Can't create inner file %q: %v", tmpdir2path, err)
+	}
+	cleanupFuncs = append(cleanupFuncs, func() { tmpdir2innerfile.Close() })
+
+	if _, err := tmpdir2innerfile.WriteString(tmpdir2innerfilecontent); err != nil {
+		executeFuncsReverse(cleanupFuncs)
+		t.Fatalf("Can't write to file %q after mounting: %v", tmpdir2innerfile, err)
+	}
+
+	return cleanupFuncs
+}
+
+type volumeMountTestCase struct {
+	description string
+	// [image name]:[image patches]
+	images         []imagePatch
+	cmdArgs        string
+	podManifest    *schema.PodManifest
+	expectedExit   int
+	expectedResult string
+}
+
+var (
+	volumeMountTestCasesRecursiveCLI = []volumeMountTestCase{
+		{
+			"CLI: recursive mount read file",
+			[]imagePatch{
+				{
+					"rkt-test-run-read-file.aci",
+					[]string{fmt.Sprintf("--exec=/inspect --read-file --file-name %s", tmpdir2filepathpod)},
+				},
+			},
+			fmt.Sprintf(
+				"--volume=test1,kind=host,source=%s,recursive=true --mount volume=test1,target=%s",
+				tmpdir, tmpdirpathpod,
+			),
+			nil,
+			0,
+			tmpdir2innerfilecontent,
+		},
+	}
+
+	volumeMountTestCasesNonRecursiveCLI = []volumeMountTestCase{
+		{
+			"CLI: read file with non-recursive mount",
+			[]imagePatch{
+				{
+					"rkt-test-run-read-file.aci",
+					[]string{fmt.Sprintf("--exec=/inspect --read-file --file-name %s", tmpdir2filepathpod)},
+				},
+			},
+			fmt.Sprintf(
+				"--volume=test,kind=host,source=%s,recursive=false --mount volume=test,target=%s",
+				tmpdir, tmpdirpathpod,
+			),
+			nil,
+			0,
+			tmpdir2outerfilecontent,
+		},
+	}
+
+	volumeMountTestCasesRecursivePodManifest = []volumeMountTestCase{
+		{
+			"Read of nested file for recursive mount",
+			[]imagePatch{
+				{"rkt-test-run-pod-manifest-recursive-mount-stat.aci", []string{}},
+			},
+			"",
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", tmpdir2filepathpod},
+							},
+							MountPoints: []types.MountPoint{
+								{tmpdirmountname, tmpdirpathpod, false},
+							},
+						},
+					},
+				},
+				Volumes: []types.Volume{
+					{Name: tmpdirmountname, Kind: "host", Source: tmpdir,
```

Comment:
IMO one line per field + the `{` on its own line would improve readability. But I guess it's a matter of taste.